CMakeLists.txt

```diff
@@ -9,11 +9,37 @@ function(get_versions versionstring VERSION_MAJOR VERSION_MINOR VERSION_PATCH)
     set(VERSION_PATCH ${numbers} PARENT_SCOPE)
 endfunction()
 
-if(NOT EXISTS "${CMAKE_SOURCE_DIR}/version.txt")
-  message(FATAL_ERROR "version.txt file does not exist")
-endif()
+execute_process(
+    COMMAND git describe --abbrev=0 --tags
```

Comment:
change to:
`COMMAND git describe --match "v[0-9]*.[0-9]*.[0-9]*" --abbrev=0 --tags`
`WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}`